cpp/src/gandiva/bitmap_accumulator.cc

```diff
@@ -28,15 +28,30 @@ void BitMapAccumulator::ComputeResult(uint8_t* dst_bitmap) {
     // set all bits to 0.
     memset(dst_bitmap, 0, arrow::BitUtil::BytesForBits(num_records));
   } else {
-    IntersectBitMaps(dst_bitmap, src_maps_, num_records);
+    IntersectBitMaps(dst_bitmap, src_maps_, src_map_offsets_, num_records);
+  }
+}
+
+void BitmapAnd(uint8_t* left, int64_t left_offset, uint8_t* right, int64_t right_offset,
+               int64_t length, uint8_t* out) {
+  if (left_offset == 0 && right_offset == 0) {
```

Comment:
arrow::internal::BitmapAnd already has a special case for this (AlignedBitmapOp). can we depend on that instead ?

Reply:
Yeah, that would be easier. The only reason I didn't use that is that the `AlignedBitmapOp` uses one byte as a time, while the previous implementation in `BitmapAccumulator` uses 8-byte at a time. If we switch to use one byte, I am not sure if the performance would degrade on certain platforms. Or maybe it doesn't matter that much?

Reply:
okay, this is perf sensitive code and the 8-byte alignment is a common case (all of java does that). I think it's good to optimise for it - so, let's keep it.